src/mocks/api/shopsResult.ts

```diff
@@ -0,0 +1,31 @@
+import { http, HttpResponse } from 'msw';
+import { API_PATH } from 'config/constants';
+import shopsResultData from './data/shopsResultData';
+
+interface RequestBody {
+  lat: number | undefined;
+  lng: number | undefined;
+}
```

Comment:
undefined 를 꼭 써야만 했던 이유가 있을까요?
api 자체가 둘 중 한개가 undefined 라면 500 에러를 내보내기 때문에 애초에 undefined인 것을 타입 단에서 방지하면 어떨까요?

<img width="1400" alt="image" src="https://github.com/BCSDLab/JJBAKSA_FRONT_END/assets/51052049/4287b14e-e5c3-4426-bde9-9dfee24a1ca7">

```suggestion
interface RequestBody {
  lat: number;
  lng: number;
}
```

만약 현재와 같은 형태를 유지하려 한다면 아래와 같이는 못 가져갈까요?

```suggestion
interface RequestBody {
  lat?: number;
  lng?: number;
}
```

이후 외부 fetch 로직에서 보내주는 body에 lat이나 lng이 undefined이라면 RequestBody 타입에 맞지 않기 때문에 mocking된 api에서는 500에러를 보내주게끔 설계하면 될 것 같네요!

혹여 500 에러에 대한 ResponseBody를 구현하신다고 하면 다음과 같은 형태로 가져가면 될 것 같습니다.

<img width="1312" alt="image" src="https://github.com/BCSDLab/JJBAKSA_FRONT_END/assets/51052049/6a7af9f6-a708-4af2-9f01-2663da793191">

Reply:
`useGeolocation`을 사용하려다 하드 코딩으로 변경한 것이었는데 그 과정에서 undefined가 남아버렸습니다. 하단의 리뷰처럼 server api와 유사한 형식으로 변경되게 `useGeolocation`을 활용하겠습니다!

Reply:
무의식적으로 useGeolocation을 떠올려서 타입에 undefined도 추가했었네요. 타입을 단언하는 방식으로 변경하겠습니다.